### PR TITLE
Unified agent diagnose command line interface

### PR DESCRIPTION
**Pull request by iglendd**
### What does this PR do?
Complements previous extended `agent diagnose` command PRs
* [First cut of agent diagnose CLI command extension #16719](https://github.com/DataDog/datadog-agent/pull/16719))
* [Move agent diagnose execution into agent process context #18792](https://github.com/DataDog/datadog-agent/pull/18792))

... and unifies previous "old" and "new" CLI interface (per discussion in [PR 16719](https://github.com/DataDog/datadog-agent/pull/16719) with @hush-hush and @Kaderinho). That pull request has added a new `all` command. This PR removes
* `all`
* `datadog-connectivity`
* `metadata-availability`

diagnose sub-commands. First one becomes synonym of simply `diagnose` command and last two into dedicated `datadog-connectivity` and `metadata-availability` diagnose suites. 

Note: Current list of agent diagnose suites
```
"C:\Program Files\Datadog\Datadog Agent\bin\agent.exe" diagnose --list --local
Diagnose suites ...
  1. check-datadog
  2. connectivity-datadog-autodiscovery
  3. connectivity-datadog-core-endpoints
  4. connectivity-datadog-event-platform
```

### Motivation
Have a single CLI interface

### Additional Notes
1. There is one more agent diagnose subcommand - `show-metadata` (which has its own two subcommands). This functionality will be moved to a dedicated (and future) `agent telemetry` command.
2. `agent flare` command is extended to run and save results of a single and full `agent diagnose` command

### Describe how to test/QA your changes
1. Makes sure that `agent diagnose show-metadata` subcommand (with all of its own subcommands and options is still working).
2. Make sure that `agent flare` command output contain agent diagnose output file.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: pkg/diagnose/connectivity/hooks.go [hunk 1]
```diff
@@ -29,9 +26,12 @@ import (
 // During a request, the http.Client will call the functions of the ClientTrace at specific moments
 // This is useful to get extra information about what is happening and if there are errors during
 // connection establishment, DNS resolution or TLS handshake for instance
-func createDiagnoseTrace(writer io.Writer) *httptrace.ClientTrace {
+func createDiagnoseTraces(httpTraces []string) *httptrace.ClientTrace {
```
Comment (hush-hush) [inline pkg/diagnose/connectivity/hooks.go:29]: The comment need to be updated to match the new function name.

File: pkg/diagnose/connectivity/hooks.go [hunk 2]
```diff
@@ -53,45 +53,40 @@ func createDiagnoseTrace(writer io.Writer) *httptrace.ClientTrace {
...
-	// color functions are defined here to keep consistency when displaying information as it might be
-	// easier to read all the information about a specific section when the information concerning this
-	// section are colored with the same color.
-	dnsColorFunc  = color.MagentaString
-	tlsColorFunc  = color.YellowString
-	hintColorFunc = color.CyanString
-)
-
-// writeWrapper is used to pass a io.Writer variable to the hooks so that they
+// writeWrapper is used to pass slice a io.Writer variable to the hooks so that they
```
Comment (hush-hush) [inline pkg/diagnose/connectivity/hooks.go:56]: Is this comment still valid ? Seems to talk about something else.

File: releasenotes/notes/diagnose-unified-cli-7214d7070cea3087.yaml [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+# it does not depend on any information only available in another
+# section. This may mean repeating some details, but each section
+# must be readable independently of the other.
+#
+# Each section note must be formatted as reStructuredText.
+---
+enhancements:
+  - |
+    Unified `agent diagnose` command CLI which includes "traditional"
+    and more recently added behavior.
```
Comment (hush-hush) [inline releasenotes/notes/diagnose-unified-cli-7214d7070cea3087.yaml:12]: We should expend on this for customers and list what exactly change in the CLI usage (command rename and such). Also, if we broke backward compatibility with the command name we should add a note in the upgrade section.

**Review by hush-hush — APPROVED** [review]
Added a few comment about details but looks good overall. Feel free to merge after answering them.

File: pkg/diagnose/connectivity/hooks.go (hunk 1 above)
Comment (iglendd) [reply pkg/diagnose/connectivity/hooks.go:29]: Thank you, I have updated the comment. Went through all other comments on the file and made some small adjustments as well.

File: releasenotes/notes/diagnose-unified-cli-7214d7070cea3087.yaml [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+# CHANGELOG.rst is rendered. So the text needs to be worded so that
+# it does not depend on any information only available in another
+# section. This may mean repeating some details, but each section
+# must be readable independently of the other.
+#
+# Each section note must be formatted as reStructuredText.
+---
+enhancements:
+  - |
+    Unified `agent diagnose` command CLI which includes "traditional"
```
Comment (vickenty) [inline releasenotes/notes/diagnose-unified-cli-7214d7070cea3087.yaml:11]: RST uses double backticks:
```suggestion
    Unified ``agent diagnose`` command CLI which includes "traditional"
```

File: pkg/diagnose/connectivity/hooks.go (hunk 2 above)
Comment (iglendd) [reply pkg/diagnose/connectivity/hooks.go:56]: Addressed with the changes from above comment

File: releasenotes/notes/diagnose-unified-cli-7214d7070cea3087.yaml (hunk 4 above)
Comment (iglendd) [reply releasenotes/notes/diagnose-unified-cli-7214d7070cea3087.yaml]: I have rewritten release note

**Review by alopezz — APPROVED** [review]

File: releasenotes/notes/diagnose-unified-cli-7214d7070cea3087.yaml [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+# it does not depend on any information only available in another
+# section. This may mean repeating some details, but each section
+# must be readable independently of the other.
+#
+# Each section note must be formatted as reStructuredText.
+---
+enhancements:
+  - |
+    Unified ``agent diagnose`` command CLI by removing ``all``,
+    ``datadog-connectivity`` and ``metadata-availability`` subcommands.
```
Comment (brett0000FF) [inline releasenotes/notes/diagnose-unified-cli-7214d7070cea3087.yaml:12]:
```suggestion
    ``datadog-connectivity``, and ``metadata-availability`` subcommands.
```

File: releasenotes/notes/diagnose-unified-cli-7214d7070cea3087.yaml [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+# CHANGELOG.rst is rendered. So the text needs to be worded so that
+# it does not depend on any information only available in another
+# section. This may mean repeating some details, but each section
+# must be readable independently of the other.
+#
+# Each section note must be formatted as reStructuredText.
+---
+enhancements:
+  - |
+    Unified ``agent diagnose`` command CLI by removing ``all``,
```
Comment (brett0000FF) [inline releasenotes/notes/diagnose-unified-cli-7214d7070cea3087.yaml:11]:
```suggestion
    Unified ``agent diagnose`` CLI command by removing ``all``,
```

File: releasenotes/notes/diagnose-unified-cli-7214d7070cea3087.yaml [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+    These separate subcommands became one of the diagnose suites. The
+    ``all`` subcommand became unnecessary.
+
+upgrade:
+  - |
+    ``datadog-connectivity`` and ``metadata-availability`` subcommands
+    do not exist anymore and their diagnoses are reported in a more
+    general and structured way.
+
+    Diagnostics previously reported via``datadog-connectivity``
```
Comment (brett0000FF) [inline releasenotes/notes/diagnose-unified-cli-7214d7070cea3087.yaml:22]:
```suggestion
    Diagnostics previously reported via ``datadog-connectivity``
```

File: releasenotes/notes/diagnose-unified-cli-7214d7070cea3087.yaml [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+    subcommand will be reported now as part of
+    ``connectivity-datadog-core-endpoints``suite. Correspondingly,
+    diagnostics previously reported via``metadata-availability``
```
Comment (brett0000FF) [inline releasenotes/notes/diagnose-unified-cli-7214d7070cea3087.yaml:25]:
```suggestion
    diagnostics previously reported via ``metadata-availability``
```

File: releasenotes/notes/diagnose-unified-cli-7214d7070cea3087.yaml [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+    subcommand will be reported now as part of
+    ``connectivity-datadog-autodiscovery``suite.
```
Comment (brett0000FF) [inline releasenotes/notes/diagnose-unified-cli-7214d7070cea3087.yaml:27]:
```suggestion
    ``connectivity-datadog-autodiscovery`` suite.
```

**Review by brett0000FF — APPROVED** [review]
Just suggested some minor tweaks and spacing changes.

**Comment by iglendd** [comment]
Thank you @brett0000FF for suggestions

**Review by AliDatadog — APPROVED** [review]

**Pull request merged**